src/words/TokTok.jl

```diff
@@ -0,0 +1,182 @@
+# Tok-Tok Tokenizer
+# Replace non-breaking spaces with normal spaces.
+
+const NON_BREAKING = ("\u00A0",) => " "
+
+# Pad some funky punctuation.
+const FUNKY_PUNCT_1 = string.(Tuple("،;؛¿!])}»›”؟¡%٪°±©®।॥…"))
+# Pad more funky punctuation.
+const FUNKY_PUNCT_2 = string.(Tuple("[“‘„‚«‹「『"))          
+# Pad En dash and em dash
+const EN_EM_DASHES = ("–—")
+
+# Replace problematic character with numeric character reference.
+const AMPERCENT = ("&",) => "&amp;"
+const TAB = ("\t",) => "&#9;"
+const PIPE = ("|",) => "&#124;"
+
+# Just pad problematic (often neurotic) hyphen/single quote, etc.
+const PROB_SINGLE_QUOTES = ("'", "’", "`")           
+# Group ` ` stupid quotes ' ' into a single token.
+const STUPID_QUOTES_1 = ("` `",) => "``"
+const STUPID_QUOTES_2 = ("' '",) => "''"
+
+# This is the \p{Open_Punctuation} from Perl's perluniprops
+# see http://perldoc.perl.org/perluniprops.html          
+const OPEN_PUNCT =(                 
+        "\u0f3a", "\u0f3c", "\u169b", "\u201a", "\u201e", "\u2045", "\u207d", "\uff62",
+        "\u208d", "\u2329", "\u2768", "\u276a", "\u276c", "\u276e", "\u2770", "\u2772",
+        "\u2774", "\u27c5", "\u27e6", "\u27e8", "\u27ea", "\u27ec", "\u27ee", "\u2983",
+        "\u2985", "\u2987", "\u2989", "\u298b", "\u298d", "\u298f", "\u2991", "\u2993",
+        "\u2995", "\u2997", "\u29d8", "\u29da", "\u29fc", "\u2e22", "\u2e24", "\u2e26",
+        "\u2e28", "\u3008", "\u300a", "\u300c", "\u300e", "\u3010", "\u3014", "\u3016",
+        "\u3018", "\u301a", "\u301d", "\ufd3e", "\ufe17", "\ufe35", "\ufe37", "\ufe39",
+        "\ufe3b", "\ufe3d", "\ufe3f", "\ufe41", "\ufe43", "\ufe47", "\ufe59", "\ufe5b",
+        "\ufe5d", "\uff08", "\uff3b", "\uff5b", "\uff5f")
+    
+# This is the \p{Close_Punctuation} from Perl's perluniprops            
+const CLOSE_PUNCT =(
+        "\u0f3b", "\u0f3d", "\u169c", "\u2046", "\u207e", "\u208e", "\u232a", "\uff63",
+        "\u2769", "\u276b", "\u276d", "\u276f", "\u2771", "\u2773", "\u2775", "\u27c6",
+        "\u27e7", "\u27e9", "\u27eb", "\u27ed", "\u27ef", "\u2984", "\u2986", "\u2988",
+        "\u298a", "\u298c", "\u298e", "\u2990", "\u2992", "\u2994", "\u2996", "\u2998",
+        "\u29d9", "\u29db", "\u29fd", "\u2e23", "\u2e25", "\u2e27", "\u2e29", "\u3009",
+        "\u300b", "\u300d", "\u300f", "\u3011", "\u3015", "\u3017", "\u3019", "\u301b",
+        "\u301e", "\u301f", "\ufd3f", "\ufe18", "\ufe36", "\ufe38", "\ufe3a", "\ufe3c",
+        "\ufe3e", "\ufe40", "\ufe42", "\ufe44", "\ufe48", "\ufe5a", "\ufe5c", "\ufe5e",
+        "\uff09", "\uff3d", "\uff5d", "\uff60")
+    
+# This is the \p{Close_Punctuation} from Perl's perluniprops             
+const CURRENCY_SYM = (
+        "\u00a2", "\u00a3", "\u00a4", "\u00a5", "\u058f", "\u060b", "\u09f2", "\u09f3", 
```

Comment:
Here the currency symbols in format of `\xa2` was giving some trouble so they are replace by their unicode equivalent